queries.sql

```diff
@@ -105,3 +105,79 @@ WHERE o.full_name = 'Dean Winchester' AND (a.escape_attempts IS NULL OR a.escape
 -- Who owns the most animals?
 SELECT o.full_name, COUNT(*) AS animal_count FROM owners o JOIN animals a ON o.id = a.owner_id
 GROUP BY o.full_name ORDER BY animal_count DESC LIMIT 1;
+
+
+
+
+-- JOIN TABLE ~ Queries -------------------------------
+
+
+-- Who was the last animal seen by William Tatcher?
+SELECT a.name AS animal_name FROM animals a
+INNER JOIN visits v ON v.animal_id = a.id
+INNER JOIN vets vt ON vt.id = v.vet_id
+WHERE vt.name = 'William Tatcher'
+ORDER BY v.visit_date DESC LIMIT 1;
+
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT v.animal_id) AS total_animals FROM visits v
+INNER JOIN vets vt ON vt.id = v.vet_id
+WHERE vt.name = 'Stephanie Mendez';
+
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT v.name AS vet_name, COALESCE(s.name, 'No Specialty') AS specialty FROM vets v
+LEFT JOIN specializations sp ON sp.vet_id = v.id
+LEFT JOIN species s ON s.id = sp.species_id;
+
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name AS animal_name FROM animals a
+INNER JOIN visits v ON v.animal_id = a.id
+INNER JOIN vets vt ON vt.id = v.vet_id
+WHERE vt.name = 'Stephanie Mendez' AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+
+-- What animal has the most visits to vets?
+SELECT animals.name FROM (
+  SELECT count (*) as num_of_visit, animal_id
+  FROM visits GROUP BY animal_id ORDER BY num_of_visit DESC LIMIT 1
+  ) 
+top_visit JOIN animals ON top_visit.animal_id = animals.id;
```

Comment:
Kindly update the screenshot result for this query as  the animal with the most visits to the vet should return `boarmon` which has 4 visit